base/sim/FairMCApplication.cxx

```diff
@@ -419,6 +419,7 @@ void FairMCApplication::FinishRun()
         UndoGeometryModifications();
     }
 
+    FairRunSim::Instance()->SetIsMT(fMC->IsMT());
```

Comment:
Could you check, whether it is possible to move this into `CloneForWorker`?

```c++
    workerRun->SetIsMT(fRun->IsMT());
```

Reply:
This change passes the actual setting of the flag from the `TVirtualMC` to the user.
Even though the user might have requested to run in the MT mode, it is not certain that this setting was actually applied (f.e. Geant4 compiled in single threaded mode, or running with TGeant3).
Therefore it cannot be called in `CloneForWorker` as this function is only called in Geant4MT.

Reply:
If we want to inform the user, what has happened really, I would propose to have a different channel from the "what the user would like to happen".

Reply:
@karabowi Would you like to comment on this?

Reply:
Is that enough?
` // Pass on the actual MT flag from the VMC to the user
    FairRunSim::Instance()->SetIsMT(fMC->IsMT());`

Reply:
Could we have a new member variable like `MTWasUsed` on FairRunSim please?

From a user point of view, it is not obvious, that FairRunSim::IsMT() before a Run gives the indended MT flag, while after the run it gives the actual one.

If you *really* want this behaviour, please write Doxygen documentation at the IsMT() method, please.

Reply:
Generally looks good, resolved.